### PR TITLE
Week 5 completed, whole TEBOW IT took around 40 minutes :D

### PR DESCRIPTION
**Pull request by loresfca**
Added two test cases on the computer run because I noticed them after I finished testing the whiteboard solution.

### CODE REVIEW

File: Week 5/Solution.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    	//value until the stack is empty or they are not the same
+    	while(!first_half.isEmpty()){
+    		char curr = first_half.pop();
+    		if(curr!=head.value) return false;
+    		head = head.next;
+    	}
+    	//if we reach this point the list is a palindrome so we return true
+    	return true;
+    }
+    public int getLength(Node head){
```
Comment (maknoon) [inline Week 5/Solution.java:43]: I like how you separated this into a different method! and good job with the really detailed comments :)

**Review by maknoon — COMMENTED** [review]
I am just for brainstorming, because I had the same idea as you to use a stack for palindrome verification, but did you figure out a way to complete the solution without using extra space/data structures? I was thinking about maybe trying to reverse the linked list by using another pointer, but was unsure how to go about implementing the verifying part for that. Any thoughts? Other than that I really like your solution, very solid and easy to understand as always!

**Comment by loresfca** [comment]
@maknoon I haven't thought of that! mmm maybe we can reverse the second half and in the reverse method return the head of the reversed part. After that, we can traverse both the first and the second half and if we get to a point where they are not the same return false but we would have to modify the list and I'm not sure if we can do that. I'll try to implement it and add it as a secondary solution. Thanks for the idea!

File: Week 5/Solution.java (hunk 1 above)
Comment (loresfca) [reply Week 5/Solution.java:43]: thanks!

**Comment by loresfca** [comment]
@itsnotdevin Thanks for your suggestions! I was actually struggling to find if the length of the list was even or odd without having to traverse it again. I'll try to improve Solution2 as soon as possible

File: Week 5/TEBOW IT /TEBOW.txt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+-Input: Head of the list
+-Null : NullPointerException
+-Return true/false
+-Single linked list
+(assumptions)
+-No variable with the length
+-Additional data structures allowed
+-Case sensitive
+
+Example:
```
Comment (charientism) [inline Week 5/TEBOW IT /TEBOW.txt:15]: Good examples :)

File: Week 5/TEBOW IT /TEBOW.txt [hunk 3]
```diff
@@ -0,0 +1,53 @@
+Is a Palindrome
```
Comment (charientism) [inline Week 5/TEBOW IT /TEBOW.txt:1]: Thanks for uploading your TEBOW IT notes!

File: Week 5/Solution2.java [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+public class Solution2 {
+	//Node structure considered for this problem
+	class Node{
+		char value;
+		Node next;
+		Node(char value, Node next){
+			this.value = value;
+			this.next = next;
+		}
+	}
```
Comment (charientism) [inline Week 5/Solution2.java:16]: nit: remember to leave an extra newline between different classes/methods etc.

File: Week 5/Solution2.java [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        //aux is created to avoid overriding head
+        Node aux = head;
+        int i = 0;
+        while(aux!=null){
+            i++;
+            aux=aux.next;
+        }
+        return i;
+    }
+    public Node getMiddle(Node head){
```
Comment (charientism) [inline Week 5/Solution2.java:68]: Clever :) You'll see that we use the same method to get the middle node during the session tomorrow

File: Week 5/Solution2.java [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+		}
+	}
+    public boolean isPalindrome(Node head) {
+    	//input validation
+    	if(head == null) throw new NullPointerException();
+    	//getting the middle element and length of the list to know if it is even or odd
+        int ln = getLength(head);
+    	Node middle = getMiddle(head);
+        //if odd we ignore the middle element
+    	if(ln%2!=0) middle = middle.next;
```
Comment (charientism) [inline Week 5/Solution2.java:24]: Food for thought - you can actually get away with not calculating the length at the beginning. You have almost all of the pieces needed for that solution here already :) Hint - how would using pointers moving at different speeds allow you to determine the middle element _and_ also give you the information necessary to tell if the list is odd?

File: Week 5/Solution2.java [hunk 7]
```diff
@@ -0,0 +1,109 @@
+import java.io.*;
+import java.util.*;
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
```
Comment (charientism) [inline Week 5/Solution2.java:4]: Great job with this + your original implementation :) Thanks for testing and leaving thorough comments!

**Review by charientism — APPROVED** [review]

File: Week 5/Solution2.java (hunk 4 above)
Comment (loresfca) [reply Week 5/Solution2.java:16]: Will do, thanks!

File: Week 5/Solution2.java (hunk 6 above)
Comment (loresfca) [reply Week 5/Solution2.java:24]: Thanks for the retro! I'll correct it as soon as possible :)

**Pull request merged**